rheda-server/databaseengine.cpp

```diff
@@ -0,0 +1,144 @@
+#include "databaseengine.h"
+
+#include <QSql>
+#include <QSqlQuery>
+#include <QSqlError>
+#include <QFile>
+#include <QDate>
+#include <QDebug>
+#include <QCoreApplication>
+#include <QDir>
+
+//Директивы имен таблицы, полей таблицы и базы данных
+#define DATABASE_HOSTNAME   "localhost"
+#define DATABASE_PATH "~/dataBase" //расположение БД на сервере
+
+#define TABLE "users_table"				// Название таблицы
+#define TABLE_NAME "login"				// 1 колонка
+#define TABLE_Hashed "user_name"		// 2 колонка
+#define TABLE_Email "password"			// 3 колонка
+#define TABLE_Status "status"			// 4 колонка
+#define TABLE_Session "session"			// 5 колонка
+
+DataBaseEngine::DataBaseEngine(QObject *parent) : QObject(parent) {
+	m_direction = DATABASE_PATH;
+}
+
+void DataBaseEngine::connectToDataBase() {
+	// Перед подключением к базе данных производим проверку на её существование.
+	// В зависимости от результата производим открытие базы данных или её восстановление
+
+	if (!QFile(m_direction).exists()) {
```

Comment:
```suggestion
	if (!QFile(m_direction).exists()) {
```
Зачем создавать объект когда есть статический метод?